src/Components/ComponentSystem.cs

```diff
@@ -0,0 +1,9 @@
+﻿namespace Components
+{
+	public interface ComponentSystem
+	{
+		void Assign<Entity, Component>(Entity entity, Component component);
```

Comment:
Thinking out loud, shouldn't always be `TryAssign` to keep the API consistent with `EntityState`?